projects/ad9361/src/main.c

```diff
@@ -152,17 +156,22 @@ struct axi_dac_init tx_dac_init = {
 struct axi_dmac_init rx_dmac_init = {
 	"rx_dmac",
 	CF_AD9361_RX_DMA_BASEADDR,
-	DMA_DEV_TO_MEM,
 	0
 };
 struct axi_dmac *rx_dmac;
 struct axi_dmac_init tx_dmac_init = {
 	"tx_dmac",
 	CF_AD9361_TX_DMA_BASEADDR,
-	DMA_MEM_TO_DEV,
-	DMA_CYCLIC
+	0
 };
 struct axi_dmac *tx_dmac;
+// MEM2MEM DMA
+struct axi_dmac_init mem_dmac_init = {
+	"mem_dmac",
+	0x7C600000,
```

Comment:
I'd still use macros for these raw values. Easier to understand from the user perspective.

Reply:
You are right. However, I would try to modify the main file as little as possible. An MEM to MEM transfer is unlikely to be implemented at this phase. The main provided in this DRAFT is only for providing a better way to understand the changes in the DMAC driver.